extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php

```diff
@@ -0,0 +1,26 @@
+<?php
+require_once( getenv( 'MW_INSTALL_PATH' ) !== false ? getenv( 'MW_INSTALL_PATH' ) . '/maintenance/Maintenance.php' : dirname( __FILE__ ) . '/../../../../maintenance/Maintenance.php' );
+
+class CleanupCommentsIndex extends Maintenance {
+
+
+	public function execute() {
+		$getIdsQuery = "select page_id from page, comments_index where page_id=comment_id and page_namespace not in (1,500,501,1200,1201,2000,2001)";
+
+		$db = wfGetDB( DB_MASTER );
+
+		$ids = [];
+
+		$result = $db->query( $getIdsQuery );
+		while ( $row = $db->fetchRow( $result ) ) {
+			$ids[] = intval( $row['page_id'] );
+		};
+
+		$deleteQuery = "DELETE FROM comments_index WHERE comment_id in (" . implode( ',', $ids ) . ")";
```

Comment:
And log how many rows were affected + execute `wfWaitForSlaves` to avoid replication lag when this will be run across multiple wikis via `run_maintenance`.